packages/web-components/src/components/filter-panel/filter-panel-composite.ts

```diff
@@ -412,25 +426,32 @@ class DDSFilterPanelComposite extends HostListenerMixin(
    */
   protected _renderDesktop = (): TemplateResult => html`
     <dds-filter-panel
-      heading="${this._filterButtonTitle}"
+      heading="${this._getComposedHeadingFilterCount()}"
       ?has-selections="${this._selectedValues.length}">
-      ${this._title.map((e) => {
-        return html` ${unsafeHTML((e as HTMLElement).outerHTML)} `;
-      })}
-      ${this._contents.map((e) => {
-        return html` ${unsafeHTML((e as HTMLElement).outerHTML)} `;
-      })}
+      <slot @slotchange="${this._handleSlotChange}"></slot>
     </dds-filter-panel>
   `;
 
   render() {
+    // Note that the <slot name="heading"> contents, intended to be
+    // <dds-filter-panel-heading> are never shown as is. The text contents
+    // are composed, using this._getComposedHeadingFilterCount(), together with
+    // the current filter count, and passed as an attribute to
+    // <dds-filter-panel-modal> and <dds-filter-panel>.
     return html`
-      <button class="bx--filter-button" @click=${this._openModal}>
-        <div class="${prefix}--filter__modal__button">
-          ${this._filterButtonTitle} ${Filter()}
-        </div>
-      </button>
-      ${this._renderModal()} ${this._renderDesktop()}
+      <slot
+        name="heading"
+        @slotchange="${this._handleHeadingSlotChange}"></slot>
+      ${this._isMobile
+        ? html`
+            <button class="bx--filter-button" @click=${this._openModal}>
+              <div class="${prefix}--filter__modal__button">
+                ${this._getComposedHeadingFilterCount()} ${Filter()}
+              </div>
+            </button>
+            ${this._renderModal()}
```

Comment:
Nit: I wonder if it makes sense to put this all in a `protected _renderMobile()` method, just so things match. Could make this method really tidy:
```js
return html`
  <slot>...</slot>
  ${this._isMobile ? this._renderMobile() : this._renderDesktop()}
`
```